### ISSUE TITLE
[Bug]: Links from external apps opening in Chromium when Puppeteer is still running

### ISSUE THREAD

**Issue by colinfran** [puppeteer/puppeteer#8948]
### Bug description

When running Puppeteer with Chromium with a long task, while task is still running, if you click on a link within an external app (ie Slack, Microsoft Teams, etc), the link is opening in Chromium instead of the default browser. Once Chromium closes, links start opening in default browser again. So as long as Chromium is open, ALL links seem to open in it.

If you switch the `executablePath` to use your local google chrome, this issue does not occur, and links open with the correct default browser.

Steps to reproduce the problem:
1. Run Puppeteer with long task using Chromium as executable.
2. Click on any link in an external app (ie Slack, Microsoft Teams, etc) manually
3. You should see that manually clicked link open in Chromium instead of the user's default browser

Running on MacOS Big Sur on M1 Macbook Air

### Puppeteer version

17.1.3

### Node.js version

v14.18.3

### npm version

6.14.16

### What operating system are you seeing the problem on?

macOS

**Comment by OrKoN** [comment]
I am unable to reproduce this on MacOS Monterey. Also, I don't think Chromium registers itsels as a default app so the OS configuration must be a reason for this. What's your default browser setting? Safari, I assume?

**Comment by OrKoN** [comment]
Also, does it happen if you run Chromium directly without using Puppeteer, e.g., `./node_modules/puppeteer/.local-chromium/mac-1036745/chrome-mac/Chromium.app/Contents/MacOS/Chromium --user-data-dir=/tmp/chromium`?

**Comment by colinfran** [comment]
> I am unable to reproduce this on MacOS Monterey. Also, I don't think Chromium registers itsels as a default app so the OS configuration must be a reason for this. What's your default browser setting? Safari, I assume?

I am using Brave Browser as my default browser. 

> ./node_modules/puppeteer/.local-chromium/mac-1036745/chrome-mac/Chromium.app/Contents/MacOS/Chromium --user-data-dir=/tmp/chromium

I am able to replicate issue with just Chromium, without Puppeteer. Could this be a Chromium issue?

**Comment by colinfran** [comment]
I think this issue can be closed. I have come to the conclusion that this issue is related to Brave Browser.

I tested this by Changing my default browser to Google Chrome in Mac system preferences.
Then I opened Chromium. and Clicked a link within slack. Link opened in Chrome as intended.

When I changed default browser back to Brave Browser, opened Chromium, and then clicked link within slack, it would open in Chromium.

so it seems Brave Browser is the culprit.
